mono/metadata/threads.c

```diff
@@ -1909,45 +1947,73 @@ mono_thread_internal_current (void)
 	return res;
 }
 
+static MonoInternalThreadHandle
+mono_thread_internal_current_handle (void)
+{
+	return MONO_HANDLE_NEW (MonoInternalThread, mono_thread_internal_current ());
+}
+
+
+static gboolean
+mono_join_uninterrupted_loop (MonoThreadHandle* thread_to_join, gint32 ms, gint64 start, gint32 *wait, MonoThreadInfoWaitRet *ret, MonoError *error)
+// A separate function is needed to avoid creating coop handles in a loop.
+{
+	gboolean done = TRUE;
+	MonoExceptionHandle exc;
+
+	HANDLE_FUNCTION_ENTER ();
+
+	MONO_ENTER_GC_SAFE;
+	*ret = mono_thread_info_wait_one_handle (thread_to_join, *wait, TRUE);
```

Comment:
*wait was ms -- separate matter than coop, and in another PR.

Reply:
This has been commited separately.
https://github.com/mono/mono/commit/0b52b173f566b67c0e5f2c203f35e379129e2fd0